cli/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Cumulus CLI library.
+
+#![warn(missing_docs)]
+
+use sc_cli;
+use std::{
+	fs,
+	io::{self, Write},
+};
+use structopt::StructOpt;
+
+/// The `purge-chain` command used to remove the whole chain: the parachain and the relaychain.
+#[derive(Debug, StructOpt)]
+pub struct PurgeChainCmd {
+	/// The base struct of the purge-chain command.
+	#[structopt(flatten)]
+	pub base: sc_cli::PurgeChainCmd,
+
+	/// Only delete the para chain database
+	#[structopt(long = "parachain", aliases = &["para"])]
+	pub parachain: bool,
+
+	/// Only delete the relay chain database
+	#[structopt(long = "relaychain", aliases = &["relay"])]
+	pub relaychain: bool,
+}
+
+impl PurgeChainCmd {
+	/// Run the purge command
+	pub fn run(
+		&self,
+		para_config: sc_service::Configuration,
+		relay_config: sc_service::Configuration,
+	) -> sc_cli::Result<()> {
+		let databases = match (self.parachain, self.relaychain) {
+			(true, true) | (false, false) => vec![para_config.database, relay_config.database],
+			(true, false) => vec![para_config.database],
+			(false, true) => vec![relay_config.database],
+		};
+
+		let db_paths = databases
+			.iter()
+			.map(|x| {
+				x.path().ok_or_else(|| {
+					sc_cli::Error::Input("Cannot purge custom database implementation".into())
```

Comment:
Maybe a hint which on this is, would be nice :D

Reply:
Done in 670b674